File: codeforces/mikeandfax.cpp
#include<iostream>
#include<string>
using namespace std;
int main()
{
	string a;
	int num;
        cin>>a;
	cin>>num;
        int start=0,flag=0;
        int last=num-1;
	int len=a.length();
        if((len%num)!=0) 
		flag=1;
        while(1) 
        {
		if(flag==1 || last>(len-1) || start>(len-1))
			break;
                int j=0;
                
		for(int i=start;i<start+(num+1)/2;i++)
		{
//                cout<<a.at(i)<<" "<<a.at(last-j)<<endl;
			if(a.at(i)!=a.at(last-j))
			{
				flag=1;	
				break;
			}
			j++;
		}
                start=last+1;
                last=last+num;
    //            cout<<start<<" "<<last<<endl;                
        }
	if(flag==0)
		cout<<"YES"<<endl;
	else
		cout<<"NO"<<endl;
	return 0;
}

File: codeforces/mikeandfax.py
x=raw_input("")
num=raw_input("")
num=int(num)
chunks,chunk_size=len(x),len(x)/num
strings = [ x[i:i+chunk_size] for i in range(0, chunks, chunk_size) ]
#print strings
flag="YES"
for i in strings:
    if i!=i[::-1]:
        flag="NO"
if len(x)%num!=0:
	flag="NO"
print flag
